### PR TITLE
Moving python gradient_clipping_threshold to protobuf and C++

### PR DESCRIPTION
**Pull request by reyoung**
Moving python gradient_clipping_threshold to protobuf and C++, make Paddle handle gradient_clipping_threshold correctly.

### CODE REVIEW

File: paddle/trainer/TrainerConfigHelper.cpp [hunk 1]
```diff
@@ -36,6 +36,22 @@ namespace paddle {
...
+    if (conf.default_values().has_gradient_clipping_threshold()) {
+      auto defaultGradientClipping =
+          conf.default_values().gradient_clipping_threshold();
+      for (auto &param : *conf.mutable_model_config()->mutable_parameters()) {
+        if (!param.has_gradient_clipping_threshold()) {
+          param.set_gradient_clipping_threshold(defaultGradientClipping);
+        }
+      }
+    }
+  }
```
Comment (lcy-seso) [inline paddle/trainer/TrainerConfigHelper.cpp]: how about other parameters except `gradient_clipping_threshold`? (parameters in this issue https://github.com/PaddlePaddle/Paddle/issues/2578)

Comment (lcy-seso) [inline paddle/trainer/TrainerConfigHelper.cpp]: currently, `momentum` is another example that is crucial to learning performance but is not correctly set. This is a test case https://github.com/PaddlePaddle/book/blob/develop/03.image_classification/train.py#L52.

Comment (reyoung) [inline paddle/trainer/TrainerConfigHelper.cpp]: In this PR, only `gradient_clipping_threshold` will be fixed. This PR is trying to explore a unified way to fix global variables bugs, not trying to fix them in one PR.

The reset variables would be fixed by following PRs.

Comment (lcy-seso) [inline paddle/trainer/TrainerConfigHelper.cpp]: I cannot agree with this modification.

If we are not going to remove `config_parser.py`, the modification in this PR means a user cannot check and confirm the parameters he set in defining the task by print the protostr. How can I confirm whether `gradient_clipping` is enable if PaddlePaddle updates? No test guarantee this.

Comment (reyoung) [inline paddle/trainer/TrainerConfigHelper.cpp]: You're right, maybe there should be another way.

**Pull request closed without merging**